src/core.jl

```diff
@@ -135,23 +136,32 @@ struct RecvBuf
     lcm::Ptr{Cvoid}
 end
 
-function onresponse(rbuf::RecvBuf, channelbytes::Ptr{UInt8}, opts::SubscriptionOptions{T}) where T
-    channel = unsafe_string(channelbytes)
+function check_channel_name(channelptr::Ptr{UInt8}, opts::SubscriptionOptions)
+    i = 1
+    while (byte = unsafe_load(channelptr)) != 0x00
```

Comment:
You can do `unsafe_load(channelptr, i)` here to avoid having to increment `channelptr`.

Reply:
Done.